it/tutorials/scala-for-java-programmers.md

```diff
@@ -0,0 +1,716 @@
+
+Un piccolo problema dei metodi `re` e `im`  è che, per essere invocati, è
+necessario far seguire il nome del metodo da una coppia di parentesi tonde
+vuote, come mostrato nel codice seguente:
+
+    object ComplexNumbers {
+      def main(args: Array[String]) {
+        val c = new Complex(1.2, 3.4)
+        println("imaginary part: " + c.im())
+      }
+    }
+
+Sarebbe decisamente meglio riuscire ad accedere alla parte reale ed immaginaria
+come se fossero campi senza dover scrivere anche la coppia vuota di parentesi.
+Questo è perfettamente fattibile in Scala semplicemente definendo i relativi
+metodi *senza argomenti*. Tali metodi differiscono da quelli con zero argomenti
+perché non presentano la coppia di parentesi dopo il nome nè nella loro
+definizione, nè nel loro utilizzo. La nostra classe `Complex` può essere
+riscritta come segue:
+
+    class Complex(real: Double, imaginary: Double) {
+      def re = real
+      def im = imaginary
+    }
+
+
+### Eredità e overriding
+
+In Scala tutte le classi sono figlie di una super-classe. Quando nessuna
+super-classe viene specificata, come nell’esempio della classe `Complex`,
+la classe `scala.AnyRef` è implicitamente usata.
+
+In Scala è possibile eseguire la sovrascrittura (_override_) dei metodi
+ereditati dalla super-classe. È pertanto necessario specificare esplicitamente
+il metodo che si sta sovrascrivendo usando il modificatore `override` per
+evitare sovrascritture accidentali. Come esempio estendiamo la nostra classe
+ `Complex` ridefinendo il metodo `toString` ereditato da `Object`.
+
+    class Complex(real: Double, imaginary: Double) {
+      def re = real
+      def im = imaginary
+      override def toString() =
+        "" + re + (if (im < 0) "" else "+") + im + "i"
+    }
+
+
+## Classi Case  e Pattern Matching
+
+Un tipo di struttura dati che spesso si trova nei programmi è l’albero.
+Ad esempio, gli interpreti ed i compilatori abitualmente rappresentano i
+programmi internamente come alberi. I documenti XML sono alberi e
+diversi tipi di contenitori sono basati sugli alberi, come gli alberi
+red-black.
+
+Esamineremo ora come gli alberi sono rappresentati e manipolati in Scala
+attraverso un semplice programma calcolatrice. Lo scopo del programma è
+manipolare espressioni aritmetiche molto semplici composte da somme,
+costanti intere e variabili intere. Due esempi di tali espressioni sono
+`1+2` e `(x+x)+(7+y)`.
+
+A questo punto è necessario definire il tipo di rappresentazione per
+dette espressioni e, a tale proposito, l’albero è la più naturale, con
+i nodi che rappresentano le operazioni (nel nostro caso, l’addizione) mentre
+le foglie sono i valori (costanti o variabili).
+
+In Scala questo albero è abitualmente rappresentato usando una super-classe
+astratta per gli alberi e una concreta sotto-classe per i nodi o le
+foglie. In un linguaggio funzionale useremmo un tipo dati algebrico per
+lo stesso scopo. Scala fornisce il concetto di _classi case_ (_case classes_)
```

Comment:
Ha senso tradurre `case class` in `classi case` (che è un ibrido tra italiano e inglese)?